### PR TITLE
Enable cloud-config from removable devices

### PR DESCRIPTION
**Pull request by davidcassany**
If cloud-init data is provided as the `user-data` file in an eligible device this data is copied to both `/oem/userdata.yaml` and `/oem/userdata`. Since `/oem` is an analyzed path for custom cloud-init files this will be taken into accound on later `elemental run-stage` calls.

### CODE REVIEW

**Comment by davidcassany** [comment]
I did some tests and I could not manage to get a Fat32 formatted USB stick to work... 😞 I seamlessly worked by using an ISO and attaching it as a cdrom to the image on KVM, but it did not make the trick from a USB.

So far I created an iso simply and only incuding a `user-data` file with the following contents:

```yaml
#cloud-config
write_files:
- path: /etc/sysconfig/proxy
  append: true
  content: |
    PROXY_ENABLED="yes"
    HTTP_PROXY=http://some.domain.org:8080
    HTTPS_PROXY=https://some.domain.org:8080
    NO_PROXY="localhost, 127.0.0.1"
```
And it worked fine. The ISO was done with 

```bash
mkisofs -o cidata.iso -V CIDATA -J -r user-data
```
being executed on the folder containing the `user-data` file. Note the `CIDATA` label is mandatory, this is what datasource plugin uses to consider eligible cloud-init devices. This worked fine and it was processed at Live ISO boot and also when booting the installed system. I am exploring why the FAT32 device with `CIDATA` label I created did not work.

**Comment by davidcassany** [comment]
I managed to make use of this feature by fixing the underlying datasource provider in our fork of linuxkit required changes are here https://github.com/davidcassany/linuxkit/pull/7. After merging this yip dependency should be bumped and then bump elemental-cli dependency on yip.

I am starting to believe we should definitely fork the `pkg/metadata` from linuxkit org to `rancher-sandbox` and tie yip there. So we can simply maintain this datasource as a stand alone project. This is related to https://github.com/rancher/elemental/issues/634

**Comment by davidcassany** [comment]
Blocked on https://github.com/davidcassany/linuxkit/pull/7 and eventually on https://github.com/rancher/elemental/issues/634

**Comment by kkaempf** [comment]
we will fork linuxkit to rancher-sandbox

**Comment by kkaempf** [comment]
> we will fork linuxkit to rancher-sandbox

Done. @davidcassany has admin rights 😉

**Comment by davidcassany** [comment]
Default branch of the fork has been renamed to `main` and copied to another branch called `linuxkit_original_master` to keep original branch somewhere. Then a PR against `main` has been opened including our changes and dropping all the code we are not using see https://github.com/rancher-sandbox/linuxkit/pull/1

**Comment by davidcassany** [comment]
This PR is ready to be merged, the changes required to use vFAT filesystems are already merged in rancher-sandbox/linuxkit and a PR to yip to switch to such a project is already opened too https://github.com/mudler/yip/pull/90. The code change here is not dependent on the other PRs, this is already functional for ISOs. As soon as element-cli is updated with new yip it will also support vFAT filesystems.

**Review by kkaempf — APPROVED** [review]
🤩

**Pull request merged**
